config/v1/types.go

```diff
@@ -80,6 +81,120 @@ type ConfigMapReference struct {
 	Key string `json:"filename,omitempty"`
 }
 
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Build holds cluster-wide information on how to handle builds. The canonical name is `cluster`
+type Build struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec holds user-settable values for the build controller configuration
+	// +optional
+	Spec BuildSpec `json:"spec,omitempty"`
+}
+
+type BuildSpec struct {
+	// AdditionalTrustedCA is a reference to a ConfigMap containing additional CAs that
+	// should be trusted for image pushes and pulls during builds.
+	// +optional
+	AdditionalTrustedCA ConfigMapReference `json:"additionalTrustedCA,omitempty"`
+	// BuildDefaults controls the default information for Builds
+	// +optional
+	BuildDefaults BuildDefaults `json:"buildDefaults,omitempty"`
+	// BuildOverrides controls override settings for builds
+	// +optional
+	BuildOverrides BuildOverrides `json:"buildOverrides,omitempty"`
+}
+
+type BuildDefaults struct {
+	// GitHTTPProxy is the location of the HTTPProxy for Git source
+	// +optional
+	GitHTTPProxy string `json:"gitHTTPProxy,omitempty"`
+
+	// GitHTTPSProxy is the location of the HTTPSProxy for Git source
+	// +optional
+	GitHTTPSProxy string `json:"gitHTTPSProxy,omitempty"`
+
+	// GitNoProxy is the list of domains for which the proxy should not be used
+	// +optional
+	GitNoProxy string `json:"gitNoProxy,omitempty"`
+
+	// Env is a set of default environment variables that will be applied to the
+	// build if the specified variables do not exist on the build
+	// +optional
+	Env []corev1.EnvVar `json:"env,omitempty"`
+
+	// SourceStrategyDefaults are default values that apply to builds using the
+	// source strategy.
+	// +optional
+	SourceStrategyDefaults SourceStrategyDefaults `json:"sourceStrategyDefaults,omitempty"`
+
+	// ImageLabels is a list of docker labels that are applied to the resulting image.
+	// User can override a default label by providing a label with the same name in their
+	// Build/BuildConfig.
+	// +optional
+	ImageLabels []ImageLabel `json:"imageLabels,omitempty"`
+
+	// Annotations are annotations that will be added to the build pod
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
```

Comment:
let's drop this. it was previously need to set tolerations, before tolerations became a first class field.  I don't expect anyone is using it anymore.

Reply:
should defaults have the `Tolerations` field, then?

Reply:
no, we don't support "default" tolerations in the core master-config so there would be nothing to map it to.

furthermore there's no way for a user to set tolerations on their build directly, so the idea of "override" vs "default" has no meaning.